src/hazelcore/semantics/CursorInfo.re

```diff
@@ -90,6 +100,251 @@ type t = {
   uses: option(UsageAnalysis.uses_list),
 };
 
+let is_empty_line = (cursor_term): bool => {
+  switch (cursor_term) {
+  | Line(_, line) => UHExp.is_empty_line(line)
+  | Exp(_, _)
+  | Pat(_, _)
+  | Typ(_, _)
+  | ExpOp(_, _)
+  | PatOp(_, _)
+  | TypOp(_, _)
+  | Rule(_, _) => false
+  };
+};
+
+let is_exp_inside = (cursor_term: cursor_term): bool => {
```

Comment:
instead of this, use the new parent_term part of cursor_info to figure out what is adjacent to the cursor